.github/workflows/esmeta-test262.yml

```
@@ -0,0 +1,41 @@
+name: 'esmeta test262'
+
+on: [pull_request]
+
+jobs:
+  esmeta-test262:
+    name: 'esmeta test262'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 
+        uses: actions/checkout@v3
+      - name: Setup JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
+      - name: Add to PATH
+        shell: bash
+        run: |
+          echo "ESMETA_HOME=$(pwd)/vendor/esmeta" >> $GITHUB_ENV
+      - name: download esmeta
+        run: |
+          mkdir -p "${ESMETA_HOME}"
+          git clone --branch v0.1.0-rc8 --depth 1 https://github.com/es-meta/esmeta.git "${ESMETA_HOME}"
+          cd "${ESMETA_HOME}" && git submodule update --init --depth 1
+      - name: build esmeta
+        run: |
+          cd "${ESMETA_HOME}"
+          sbt assembly
+      - name: run test262
+        run: |
+          git fetch origin main
+          paths=$(git diff --diff-filter ACMR --name-only origin/main.. -- test/)
```

If possible, it might be better to use
```suggestion
          paths=$(npm run diff)
```
in order to make sure that this script runs the same set of tests that the other jobs do. But if there's a reason not to do that (maybe we'd have to run npm install first?) then it's fine as is.

---

I wll check if it's possible later today. Thanks for great suggestion.

---

Hi, I tested out the suggested change [here](https://github.com/doehyunbaek/test262/actions/runs/3551724196/jobs/5966141508) and It did not work very well.

Actually, ci_test.sh which other jobs use, do not use `npm run diff` as well, as can be seen below.

https://github.com/tc39/test262/blob/1ebd34b53b59a5cb84bdfece8931e40c462747e0/tools/scripts/ci_test.sh#L3-L4

I could make it work by changing the value of npm run diff according to one used by ci_test.sh and this workflow, and make npm run output silent(https://github.com/doehyunbaek/test262/commit/75ae1577729f8f4a5a8890583168b99a25ce0c9e, https://github.com/doehyunbaek/test262/actions/runs/3551711422). 

If you think this is a good path forward, I will push the required changes. Also if you think ci_test.sh should also be modified to use `npm run diff`, let me know.

---

Oh, sorry for the bad suggestion. No need to dive into this more, I think we can just merge the PR as is. Thanks for checking it out.

---

If we wanted to do this in the future, I think we'd have to do something like `git checkout -b main origin/main`.